jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamInitSelectionPanel.kt

```diff
@@ -0,0 +1,169 @@
+                panel.border = IdeBorderFactory.createTitledBorder(it.title(), false)
+            }
+            panel.add(it.component(), BorderLayout.CENTER)
+            panel
+        }
+
+        fragments.setContent(
+            panel {
+                wizardFragments.values.forEach {
+                    row() {
+                        it(grow)
+                    }
+                }
+            }
+        )
+
+        runtimeUpdate()
+        wizardUpdate()
+    }
+
+    fun setRuntime(runtime: Runtime) {
+        val itemCount = runtimeComboBox.itemCount
+        for (itemIndex in 0 until itemCount) {
+            if (runtimeComboBox.getItemAt(itemIndex) == runtime) {
+                runtimeComboBox.selectedItem = runtime
+                return
+            }
+        }
+    }
+
+    private fun runtimeUpdate() {
+        val selectedRuntime = runtimeComboBox.selectedItem as? Runtime
+        templateComboBox.removeAllItems()
+
+        // if selected runtimeComboBox is null, we're on an unsupported platform
+        if (selectedRuntime == null) {
+            return
+        }
+        SamProjectTemplate.supportedTemplates()
+            .filter { it.supportedRuntimes().contains(selectedRuntime) }
+            .forEach { templateComboBox.addItem(it) }
+    }
+
+    private fun wizardUpdate() {
+        val selectedRuntime = runtimeComboBox.selectedItem as? Runtime
+        val selectedTemplate = templateComboBox.selectedItem as? SamProjectTemplate
+        wizardFragments.forEach { (wizardFragment, jComponent) ->
+            wizardFragment.updateUi(projectLocation, selectedRuntime, selectedTemplate)
+            jComponent.isVisible = wizardFragment.isApplicable(selectedTemplate)
+        }
+    }
+
+    fun validate(): ValidationInfo? {
+        val samExecutable = ExecutableManager.getInstance().getExecutableIfPresent(getExecutable(SamExecutable::class.java))
+        if (samExecutable is BadExecutable) {
+            return ValidationInfo(samExecutable.validationError, samExecutableField)
+        }
+
+        // TODO: Why does this have no validation on it?
+        val selectedRuntime = runtimeComboBox.selectedItem as? Runtime
+        if (selectedRuntime != null) {
+            val samVersion = SemVer.parseFromText(samExecutable.version)!!
+            val runtimeGroup = selectedRuntime.runtimeGroup!!
+            try {
+                runtimeGroup.validateSamVersion(selectedRuntime, samVersion)
+            } catch (e: Exception) {
+                return ValidationInfo(e.message!!, runtimeComboBox)
+            }
+        }
+
+        // TODO: Why does this have no validation on it?
+        val samProjectTemplate = templateComboBox.selectedItem as? SamProjectTemplate
+
+        return wizardFragments.keys
+            .filter { it.isApplicable(samProjectTemplate) }
+            .mapNotNull { it.validateFragment() }
+            .firstOrNull()
+    }
+
+    fun registerValidators() {
```

Comment:
is the validation ordering comment re: "performing validation only after both the step and peer ui exist" still valid here or can we throw out the `registerValidators` concept with the redo of how validation works

Reply:
This function doesn't do anything anymore due to we don't use the fake Pycharm step, but I haven't reworked our validation logic yet